src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php

```diff
@@ -302,6 +302,74 @@ public function iAddActionThatGivesFixedDiscount(string $discount, ChannelInterf
         $this->formElement->specifyLastActionDiscountForChannel($discount, $channel);
     }
 
+    /**
+     * @When I create a new catalog promotion :promotion with priority :priority that reduces price by ":discount%" and applies on :variant variant
+     */
+    public function iCreateANewCatalogPromotionWithPriorityThatReducesPriceByAndAppliesOnVariant(string $promotion, int $priority, string $discount, ProductVariantInterface $variant): void
+    {
+        $this->createPage->open();
+        $this->formElement->nameIt($promotion);
+        $this->formElement->prioritizeIt($priority);
+        $this->iAddActionThatGivesPercentageDiscount($discount);
+        $this->iAddScopeThatAppliesOnVariants($variant);
+        $this->createPage->create();
+    }
+
+    /**
+     * @When /^I create an exclusive catalog promotion with "([^"]+)" code and "([^"]+)" name and ([^"]+) priority that applies on ("[^"]+" product) and reduces price by "((?:\d+\.)?\d+)%" in ("[^"]+" channel)$/
+     */
+    public function iCreateAnExclusiveCatalogPromotionWithCodeAndPriorityThatReducesPriceByInTheChannelAndAppliesOnProduct(
+        string $code,
+        string $promotion,
+        int $priority,
+        ProductInterface $product,
+        string $discount,
+        string $channel
+    ): void
+    {
+        $this->createPage->open();
+        $this->createPage->specifyCode($code);
+        $this->formElement->labelIt($promotion, "en_US");
+        $this->formElement->nameIt($promotion);
+        $this->formElement->prioritizeIt($priority);
+        $this->formElement->checkExclusive();
+        $this->formElement->checkChannel($channel);
+        $this->formElement->addScope();
+        $this->formElement->chooseScopeType('For product');
+        $this->formElement->chooseLastScopeCodes([$product->getCode()]);
+        $this->formElement->addAction();
+        $this->formElement->chooseActionType('Percentage discount');
+        $this->formElement->specifyLastActionDiscount($discount);
+        $this->createPage->create();
```

Comment:
Can we move it to a private method? It is almost the same as the body of the next function.

Reply:
It needs to have `checkExclusive()` method inside and the other not. If we move it to to a private method it will not be the same logic in my opinion, we will make it then to 2 or 3 steps instead of 1 on each, for example when we decide to move creation of basic catalog promotion we will face that we should use `$this->createPage->create()` and `checkExclusive()` outside this private method